packages/@aws-cdk/assertions/README.md

```diff
@@ -299,6 +299,35 @@ target array. Out of order will be recorded as a match failure.
 Alternatively, the `Match.arrayEquals()` API can be used to assert that the target is
 exactly equal to the pattern array.
 
+### String Matchers
+
+The `Match.stringLikeRegexp()` API can be used to assert that the target matches the
+provided regular expression.
+
+```ts
+// Given a template -
+// {
+//   "Resources": {
+//     "MyBar": {
+//       "Type": "Foo::Bar",
+//       "Properties": {
+//         "Template": "const includeHeaders = true;"
+//       }
+//     }
+//   }
+// }
+
+// The following will NOT throw an assertion error
+template.hasResourceProperties('Foo::Bar', {
+  Fred: Match.stringLike('.*includeHeaders = true.*'),
+});
+
+// The following will throw an assertion error
+template.hasResourceProperties('Foo::Bar', {
+  Fred: Match.stringLike('.*includeHeaders = false.*'),
```

Comment:
`.*` is not necessary if the pattern doesn't start with a `^`. 

Potentially there is a use for an option which says `{ fullString: true }`, which automatically adds a `^` at the start and a `$` at the end.